viral/cache_2p_sessions.py

```diff
@@ -147,9 +147,55 @@ def add_daq_times_to_trial(
         )
 
 
+def extract_frozen_wheel_chunks(
+    stack_lengths_tiffs: np.ndarray,
+    chunk_lengths_daq: np.ndarray,
+    frame_times_daq: np.ndarray,
+    behaviour_times: np.ndarray,
+) -> tuple[tuple[int, int], tuple[int, int]]:
+    """Extract start and end frame index for pre-training and post-training imaging chunks"""
+    frame_times = list()
+    start_idx = 0
+    for chunk_len in chunk_lengths_daq:
+        frame_times.append(frame_times_daq[start_idx : start_idx + chunk_len])
+        start_idx += chunk_len
+    # first chunk (before behavioural chunks)
+    first_chunk_len = stack_lengths_tiffs[0]
+    first_chunk = (0, first_chunk_len - 1)
+
+    # last chunk (after all behavioural chunks)
+    last_chunk_len = stack_lengths_tiffs[-1]
+    prev_frames_total = sum(stack_lengths_tiffs[:-1])
+    last_chunk = (prev_frames_total - 1, prev_frames_total + last_chunk_len - 1)
+
+    # We want to make sure the chunks are >= 15 mins but < 20 mins
+    # 15 mins = 27,000 frames
+    # 20 mins = 36,000 frames
+    assert (
+        27000 <= first_chunk_len < 36000
+    ), "First chunk length does not match expected length"
+
+    assert (
+        27000 <= last_chunk_len < 36000
+    ), "Last chunk length does not match expected length"
+
+    assert not np.any(
+        np.isin(behaviour_times, frame_times[0])
+    ), "Behavioral pulses detected in pre-training period!"
+
+    assert not np.any(
+        np.isin(behaviour_times, frame_times[-1])
+    ), "Behavioral pulses detected in post-training period!"
+    return first_chunk, last_chunk
```

Comment:
This would only error if there's a frame at exactly the same time as a spacer, you need to check that there's no behaviour times at any point during the first and last chunk